File: AndroidAppBaseLibrary/src/main/java/stark/android/appbase/base/adapter/SimpleFragmentPagerAdapter.java
package stark.android.appbase.base.adapter;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by jihongwen on 16/7/28.
 */

public class SimpleFragmentPagerAdapter extends FragmentPagerAdapter {

    private List<FragmentTab> fragmentTabs;

    public SimpleFragmentPagerAdapter(FragmentManager fm, List<FragmentTab> fragmentTabs) {
        super(fm);
        this.fragmentTabs = fragmentTabs;
    }

    public SimpleFragmentPagerAdapter(FragmentManager fm) {
        super(fm);
        this.fragmentTabs = new ArrayList<>();
    }

    public void setFragmentTabs(List<FragmentTab> fragmentTabs) {
        this.fragmentTabs = fragmentTabs;
        notifyDataSetChanged();
    }

    public List<FragmentTab> getFragmentTabs() {
        return fragmentTabs;
    }

    @Override
    public Fragment getItem(int position) {
        return fragmentTabs.get(position).fragment;
    }

    @Override
    public int getCount() {
        return fragmentTabs.size();
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return fragmentTabs.get(position).tab;
    }

    public static class FragmentTab {
        public String tab;
        public Fragment fragment;

        public FragmentTab(String tab, Fragment fragment) {
            this.tab = tab;
            this.fragment = fragment;
        }
    }
}

File: AndroidAppBaseLibrary/src/main/java/stark/android/appbase/viewpager/adapter/SimpleFragmentPagerAdapter.kt
package stark.android.appbase.viewpager.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import java.util.ArrayList

/**
 * Created by jihongwen on 16/7/28.
 */

class SimpleFragmentPagerAdapter : FragmentPagerAdapter {

    private var fragmentTabs: List<FragmentTab>? = null

    constructor(fm: FragmentManager, fragmentTabs: List<FragmentTab>) : super(fm) {
        this.fragmentTabs = fragmentTabs
    }

    constructor(fm: FragmentManager) : super(fm) {
        this.fragmentTabs = ArrayList()
    }

    fun setFragmentTabs(fragmentTabs: List<FragmentTab>) {
        this.fragmentTabs = fragmentTabs
        notifyDataSetChanged()
    }

    fun getFragmentTabs(): List<FragmentTab>? {
        return fragmentTabs
    }

    override fun getItem(position: Int): Fragment {
        return fragmentTabs!![position].fragment
    }

    override fun getCount(): Int {
        return fragmentTabs!!.size
    }

    override fun getPageTitle(position: Int): CharSequence {
        return fragmentTabs!![position].tab
    }

    class FragmentTab(var tab: String, var fragment: Fragment)
}
